src/app/market/buy/checkout-process/checkout-process.component.ts

```diff
@@ -374,6 +375,17 @@ export class CheckoutProcessComponent implements OnInit, OnDestroy {
     this.stepper.linear = false;
   }
 
+  openListing(listing: any) {
+    if (new Date().getTime() < listing.listing.expiredAt) {
```

Comment:
Rather than duplicating the expiry check here, use the `this.checkExpired()` function (which does this check already)